SYCL/Basic/buffer/buffer.cpp

```diff
@@ -532,174 +539,190 @@ int main() {
       sycl::buffer<int, dims> buf_int_shrd(
           int_shrd, r,
           sycl::property_list{sycl::property::buffer::use_mutex(m)});
+#ifdef ENABLE_FP64
       sycl::buffer<double, dims> buf_double_shrd(
           double_shrd, r,
           sycl::property_list{sycl::property::buffer::use_mutex(m)});
+#endif
       m.lock();
       std::fill(bool_shrd.get(), (bool_shrd.get() + size), bool());
       std::fill(int_shrd.get(), (int_shrd.get() + size), int());
+#ifdef ENABLE_FP64
       std::fill(double_shrd.get(), (double_shrd.get() + size), double());
+#endif
       m.unlock();
-
       buf_bool_shrd.set_final_data(bool_vector.begin());
       buf_int_shrd.set_final_data(int_vector.begin());
+#ifdef ENABLE_FP64
       buf_double_shrd.set_final_data(double_vector.begin());
+#endif
+
       buf_bool_shrd.set_write_back(true);
       buf_int_shrd.set_write_back(true);
+#ifdef ENABLE_FP64
       buf_double_shrd.set_write_back(true);
+#endif
 
       Queue.submit([&](sycl::handler &cgh) {
         auto Accessor_bool =
             buf_bool_shrd.get_access<sycl::access::mode::write>(cgh);
         auto Accessor_int =
             buf_int_shrd.get_access<sycl::access::mode::write>(cgh);
+#ifdef ENABLE_FP64
         auto Accessor_double =
             buf_double_shrd.get_access<sycl::access::mode::write>(cgh);
+#endif
         cgh.parallel_for<class FillBuffer>(r, [=](sycl::id<1> WIid) {
           Accessor_bool[WIid] = true;
           Accessor_int[WIid] = 3;
+#ifdef ENABLE_FP64
           Accessor_double[WIid] = 7.5;
+#endif
         });
       });
     } // Data is copied back
 
     for (size_t i = 0; i < size; i++) {
-      if (bool_vector[i] != true || int_vector[i] != 3 ||
-          double_vector[i] != 7.5) {
-        assert(false && "Data was not copied back");
-        return 1;
+      if (bool_vector[i] != true || int_vector[i] != 3) {
+#ifdef ENABLE_FP64
+        if (bool_vector[i] != true || int_vector[i] != 3 ||
+            double_vector[i] != 7.5) {
+#endif
```

Comment:
Maybe just me, but this is hard to read. Also, I think there is a functional change here. Consider the case when bool/int are ok and double is wrong. We won't even take the first if so the double check will not be done at all. I'd change like this:

```
  bool Passed = true;
  Passed &= (bool_vector[i] == true);
  Passed &= (int_vector[i] == 3);
  #ifdef ENABLE_FP64
  Passed &= (double_vector[i] == 7.5);
  #endif
```